src/modules/item/item.dao.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Item } from '../../../entities/Item.entity';
+
+@Injectable()
+export class ItemDao {
+  constructor(
+    @InjectRepository(Item)
+    private readonly repository: Repository<Item>
+  ) {}
+
+  async getItemById(itemId: string): Promise<Item> {
+    return this.repository
+      .createQueryBuilder('item')
+      .where('item.id = :itemId', { itemId })
```

Comment:
Add a clause that based in the logged in company it gets the item (add .andWhere('item.companyId = :companyId', { companyId })), you can access the logged in company in the Users object store in the request session.

Reply:
But this should be passed as a parameter at this point, right @matheo97 ? so far that info will be retrieved in a higher level or am I wrong?

Reply:
You should get the companyId as a param and use it in a where clause